src/gensxg.c

```diff
@@ -0,0 +1,334 @@
+  }
+  if (opt->validity_url == NULL) {
+    fprintf(stderr, " -validityUrl must not be empty.\n");
+    valid = false;
+  }
+  if (opt->content == NULL) {
+    fprintf(stderr, " -content file must be specified.\n");
+    valid = false;
+  } else {
+    if (access(opt->content, F_OK)) {
+      fprintf(stderr, " Cannot access content file: %s\n", opt->content);
+      valid = false;
+    }
+  }
+  if (opt->certificate == NULL) {
+    if (opt->public_key == NULL) {
+      fprintf(stderr,
+              " -certificate file or -publicKey file must be specified.\n");
+      valid = false;
+    } else if (access(opt->public_key, F_OK)) {
+      fprintf(stderr, " Cannot access publicKey file: %s", opt->public_key);
+      valid = false;
+    }
+  } else {
+    if (opt->public_key != NULL) {
+      fprintf(stderr, " -certificate and -publicKey cannot be specified "
+              "at the same time.\n");
+      valid = false;
+    } else if (access(opt->certificate, F_OK)) {
+      fprintf(
+          stderr, " Cannot access certificate file: %s\n", opt->certificate);
+      valid = false;
+    }
+  }
+  if (opt->private_key == NULL) {
+    fprintf(stderr, " -privateKey file must be specified.\n");
+    valid = false;
+  } else {
+    if (access(opt->private_key, F_OK)) {
+      fprintf(stderr, " Cannot access privateKey file: %s\n", opt->private_key);
+      valid = false;
+    }
+  }
+  return valid;
+}
+
+static void load_signer(const Options* opt, sxg_signer_list_t* signers) {
+  if (opt->certificate) {
+    X509* cert = load_x509_cert(opt->certificate);
+    EVP_PKEY* pri_key = load_private_key(opt->private_key);
+    if (!sxg_add_ecdsa_signer(opt->url, opt->date, opt->expire,
+                              opt->validity_url, pri_key,
+                              cert, opt->cert_url, signers)) {
+      fprintf(stderr,
+          "Failed to generate SXG signer with certificate\n");
+      exit(1);
+    }
+    X509_free(cert);
+    EVP_PKEY_free(pri_key);
+  } else {
+    EVP_PKEY* pub_key = load_ed25519_pubkey(opt->public_key);
+    EVP_PKEY* pri_key = load_private_key(opt->private_key);
+    if (!sxg_add_ed25519_signer(opt->url, opt->date, opt->expire,
+                                opt->validity_url, pri_key,
+                                pub_key, signers)) {
+      fprintf(stderr, "Failed to generate SXG signer with Ed25519 key");
+      exit(1);
+    }
+    EVP_PKEY_free(pub_key);
+    EVP_PKEY_free(pri_key);
+  }
+}
+
+static void load_contents(const Options* opt, sxg_buffer_t* buf) {
+  FILE* file = fopen(opt->content, "rb");
+  if (file == NULL) {
+    fprintf(stderr, "Failed to load content: %s\n", opt->content);
+    exit(1);
+  }
+  fseek(file, 0, SEEK_END);
+  size_t filesize = ftell(file);
+  rewind(file);
+  sxg_buffer_resize(filesize, buf);
+  fread(buf->data, sizeof(char), filesize, file);
+  fclose(file);
+}
+
+static void help() {
+  fprintf(stderr, kHelpMessage);
+}
+
+static void dump(const Options* opt) {
```

Comment:
`gensxg` requires so many arguments.
Passing proper arguments is quite difficult for users so that doing feedback with incomplete argument list is helpful information, I think.